kratos/input_output/vtk_output.cpp

```diff
@@ -120,104 +127,71 @@ void VtkOutput::WriteConditionsAndElements(const ModelPart &rModelPart, std::ofs
-             cell_type = 3; // line 2d or 3d
-        if (elem.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Point2D)
-             cell_type = 1;
-        if (elem.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Point3D)
-             cell_type = 1;
-        if( cell_type < 0)
-            KRATOS_ERROR<<"Modelpart contains elements with geometries for which no VTK-output is implemented!"<<std::endl;
-
-        WriteScalarDataToFile( (int)cell_type, rFileStream);
-        if (mFileFormat == VtkOutput::FileFormat::VTK_ASCII) rFileStream << "\n";
-    }
-
-    // write conditions types
-    for (const auto& condition : local_mesh.Conditions())
+template <typename TEntity>
+void VtkOutput::WriteCellType(const typename PointerVectorSet<TEntity, IndexedObject>::ContainerType& rContainer, std::ofstream& rFileStream)
+{
+    std::unordered_map<GeometryData::KratosGeometryType, int> geo_type_vtk_cell_type_map = {
+            {GeometryData::KratosGeometryType::Kratos_Triangle2D3, 5 },
+            {GeometryData::KratosGeometryType::Kratos_Quadrilateral2D4, 9 },
+            {GeometryData::KratosGeometryType::Kratos_Tetrahedra3D4, 10},
+            {GeometryData::KratosGeometryType::Kratos_Line2D2, 3},
+            {GeometryData::KratosGeometryType::Kratos_Line3D2, 3},
+            {GeometryData::KratosGeometryType::Kratos_Point2D, 1},
+            {GeometryData::KratosGeometryType::Kratos_Point3D, 1}
+    };
+    // write entity types
+    for (const auto& entity : rContainer)
     {
-        int  cell_type = -1;
-
-        if (condition.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Triangle2D3)
-             cell_type = 5; // triangle 2d
-        if (condition.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Quadrilateral2D4)
-             cell_type = 9; // quad 2d
-        if (condition.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Tetrahedra3D4)
-             cell_type = 10; // tetra 3d
-        if (condition.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Line2D2)
-             cell_type = 3; // line 2d or 3d
-        if (condition.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Line3D2)
-             cell_type = 3; // line 2d or 3d
-        if (condition.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Point2D)
-             cell_type = 1;
-        if (condition.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Point3D)
-             cell_type = 1;
-        if( cell_type < 0)
-            KRATOS_ERROR<<"Modelpart contains elements with geometries for which no VTK-output is implemented!"<<std::endl;
+        int cell_type = -1;
+        if( geo_type_vtk_cell_type_map.count(entity->GetGeometry().GetGeometryType()) > 0  )
```

Comment:
I think usually this is done with `find`
then you can check if `result == end` and throw in this case

Up to you

Reply:
cost is the same I think !